### PR TITLE
Enable nullability in TypeCodeDomSerializer

### PR DESCRIPTION
**Pull request by halgab**
Another follow-up to #9099

## Proposed changes

- Enable nullability in `TypeCodeDomSerializer`
- Enable nullability in `CodeDomSerializationProvider`
- Enable nullability in `ComponentTypeCodeDomSerializer`

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/9529)

### CODE REVIEW

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/TypeCodeDomSerializer.cs [hunk 1]
```diff
@@ -580,18 +554,16 @@ private void IntegrateStatements(IDesignerSerializationManager manager, object r
     }
 
     #region OrderedStatementsCollection Class
-    private class StatementOrderComparer : IComparer
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/TypeCodeDomSerializer.cs:583]: you can move this to a new file.

**Review by dreddy-work — COMMENTED** [review]
This is more than nullability changes. Please rebase to .NET9.0-Test branch and we will be merging it there.

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/TypeCodeDomSerializer.cs (hunk 1 above)
Comment (halgab) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/TypeCodeDomSerializer.cs:583]: done

**Comment by halgab** [comment]
Should I rebase this PR back to main?

**Comment by halgab** [comment]
> Should I rebase this PR back to main?

I just did that, considering feature/9.0 seems to be stale @dreddy-work

**Comment by halgab** [comment]
This one has been ready for a while @lonitra

**Review by lonitra — APPROVED** [review]
LGTM

**Pull request merged**